File: jpa/spring-data-jpa/jpashop-java/src/main/java/jpastudy/jpashop/domain/item/Movie.java
package jpastudy.jpashop.domain.item;

import com.fasterxml.jackson.annotation.JsonGetter;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;

@Getter
@Setter
@Entity
@DiscriminatorValue("M")
public class Movie extends Item {

    private String director;
    private String actor;
}

File: spring-data-jpa/jpashop-kotlin/module-common/src/main/kotlin/til/dudu/common/domain/item/Movie.kt
package til.dudu.common.domain.item

import javax.persistence.DiscriminatorValue
import javax.persistence.Entity

@Entity
@DiscriminatorValue("movie")
class Movie(
    override var name: String,
    override var price: Int,
    override var stockQuantity: Int,
    val director: String,
    val actor: String
) : Item(name = name, price = price, stockQuantity = stockQuantity)